### PR TITLE
Fixes the tilde directory problem with existing layout files

### LINKED ISSUE #786: Identity scaffolding adds extra '~' directory with support files
When scaffolding identity, in addition to the files which should be created, a top-level directory named '~' is created, containing the directory `Views/Shared/` which contains some support files:
image

This happens both when an existing _Layout.cshtml file is chosen, or if a new one is being generated.

@vijayrkn @javiercn

### PR DESCRIPTION
**Pull request by seancpeters**
Fixes #786

### CODE REVIEW

**Comment by seancpeters** [comment]
@javiercn @prafullbhosale - just hoping to get a quick look at this change - it's being taken to shiproom tomorrow.

File: src/VS.Web.CG.Mvc/Identity/IdentityGeneratorTemplateModelBuilder.cs [hunk 1]
```diff
@@ -261,8 +261,22 @@ private bool DetermineSupportFileLocation(out string supportFileLocation, out st
...
+                if (_commandlineModel.Layout.StartsWith("~"))
+                {
+                    layoutFile = _commandlineModel.Layout.Substring(1);
+                }
+                else
+                {
+                    layoutFile = _commandlineModel.Layout;
+                }
+
+                if (layoutFile[0] == '\\' || layoutFile[0] == '/')
```
Comment (prafullbhosale) [inline src/VS.Web.CG.Mvc/Identity/IdentityGeneratorTemplateModelBuilder.cs]: This already handles `~/` and `~\` cases, should this support crazy input like `\\\abc\layout` or `\/abc\layout`

Comment (seancpeters) [reply src/VS.Web.CG.Mvc/Identity/IdentityGeneratorTemplateModelBuilder.cs]: Yes, it probably should. The path is assumed to always be relative to the project root, so all lead slashes should be removed. Fixing...

**Comment by seancpeters** [comment]
Closing this, it's replaced with #789

**Pull request closed without merging**